swesmith/profiles/javascript.py

```diff
@@ -315,6 +313,192 @@ def log_parser(self, log: str) -> dict[str, str]:
+RUN npm ci
+"""
+
+    def log_parser(self, log: str) -> dict[str, str]:
+        return parse_log_mocha(log)
+
+
+@dataclass
+class HighlightJS5697ae51(JavaScriptProfile):
+    owner: str = "highlightjs"
+    repo: str = "highlight.js"
+    commit: str = "5697ae5187746c24732e62cd625f3f83004a44ce"
+    test_cmd: str = "npm run test -- --verbose"
+
+    @property
+    def dockerfile(self):
+        return f"""FROM node:22-bullseye
+RUN apt update && apt install -y git
+RUN git clone https://github.com/{self.mirror_name} /testbed
+WORKDIR /testbed
+RUN npm install
+RUN npm run build
+"""
+
+    def log_parser(self, log: str) -> dict[str, str]:
+        return parse_log_mocha(log)
+
+
+@dataclass
+class Prism31b467fa(JavaScriptProfile):
+    owner: str = "PrismJS"
+    repo: str = "prism"
+    commit: str = "31b467fa7c92c5ce90c3e7c6c8fe2b8a946d9484"
+    test_cmd: str = "npm run test"
+
+    @property
+    def dockerfile(self):
+        return f"""FROM node:22-bullseye
+RUN apt update && apt install -y git
+RUN git clone https://github.com/{self.mirror_name} /testbed
+WORKDIR /testbed
+RUN npm ci
+RUN npm run build
+"""
+
+    def log_parser(self, log: str) -> dict[str, str]:
+        return parse_log_mocha(log)
+
+
+@dataclass
+class ChromaJS498427ea(JavaScriptProfile):
+    owner: str = "gka"
+    repo: str = "chroma.js"
+    commit: str = "498427eafc2e987a3751f8d5fe0612fa7a4a76ec"
+    test_cmd: str = "npm run test -- --run"
+
+    @property
+    def dockerfile(self):
+        return f"""FROM node:22-bullseye
+RUN apt update && apt install -y git
+RUN git clone https://github.com/{self.mirror_name} /testbed
+WORKDIR /testbed
+RUN npm install
+RUN npm run build
+"""
+
+    def log_parser(self, log: str) -> dict[str, str]:
+        return parse_log_vitest(log)
+
+
+@dataclass
+class Colorfef7b619(JavaScriptProfile):
+    owner: str = "Qix-"
+    repo: str = "color"
+    commit: str = "fef7b619edd678455595b9b6a10780f13b58d285"
+    test_cmd: str = "npm run test -- --verbose"
+
+    @property
+    def image_name(self) -> str:
+        # Note: "-" followed by a "_" is not allowed in Docker image names
+        return f"{self.org_dh}/swesmith.{self.arch}.{self.owner.replace('-', '_')}_1776_{self.repo}.{self.commit[:8]}".lower()
+
+    @property
+    def dockerfile(self):
+        return default_npm_install_dockerfile(self.mirror_name, node_version="22")
+
+    def log_parser(self, log: str) -> dict[str, str]:
+        return parse_log_mocha(log)
+
+
+@dataclass
+class Qd180f4a0(JavaScriptProfile):
+    owner: str = "kriskowal"
+    repo: str = "q"
+    commit: str = "d180f4a0b22499607ac750b56766c8829d6bff43"
+    test_cmd: str = "npm run test -- --verbose"
+
+    @property
+    def dockerfile(self):
+        return default_npm_install_dockerfile(self.mirror_name, node_version="22")
+
+    def log_parser(self, log: str) -> dict[str, str]:
+        return parse_log_mocha(log)
+
+
+@dataclass
+class ImmutableJS879adab5(JavaScriptProfile):
```

Comment:
As discussed in slack perhaps we don't add `immutable-js/immutable-js` as it is in SWE-bench_Multilingual.